app/src/main/java/com/belatrixsf/connect/utils/CustomDomainEditText.java

```diff
@@ -0,0 +1,131 @@
+*/
+package com.belatrixsf.connect.utils;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.support.design.widget.TextInputEditText;
+import android.text.Spannable;
+import android.text.style.ForegroundColorSpan;
+import android.util.AttributeSet;
+
+import com.belatrixsf.connect.R;
+
+/**
+ * Created by echuquilin on 24/01/17.
+ */
+public class CustomDomainEditText extends TextInputEditText {
+
+    private String defaultDomain;
+    private String defaultUsername;
+    private boolean isInitialized = false;
+    private int hintColor = getResources().getColor(R.color.grey);
+
+    public CustomDomainEditText(Context context) {
+        super(context);
+    }
+
+    public CustomDomainEditText(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public CustomDomainEditText(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+    }
+
+    @Override
+    protected void onSelectionChanged(int selStart, int selEnd) {
+        if (isInitialized) {
+            int atIndex = getText().toString().indexOf('@');
+            if (atIndex > -1) {
+                String username = getUserName();
+                if (username.equals(defaultUsername)) {
+                    if (selStart >= 0 && selEnd >= 0) {
+                        setSelection(0, 0);
+                        return;
+                    }
+                } else {
+                    if (selStart <= atIndex && selEnd >= atIndex) {
+                        setSelection(selStart, atIndex);
+                        return;
+                    } else if (selStart > atIndex && selEnd > atIndex) {
+                        setSelection(atIndex, atIndex);
+                        return;
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    protected void onTextChanged(CharSequence text, int start, int lengthBefore, int lengthAfter) {
+        if (isInitialized) {
+            String userText = getUserName();
+            if (userText.isEmpty()) {
+                setTextWithDomain(defaultUsername);
+                setFormat(hintColor);
+            } else if (!userText.equals(defaultUsername) && userText.contains(defaultUsername)) {
+                setTextWithDomain(userText.replace("Username", ""));
+                setFormat(Color.BLACK);
+            }
+        }
+    }
+
+    public void setDefaultDomain(String defaultDomain) {
+        this.defaultDomain = defaultDomain;
+    }
+
+    public String getDefaultDomain() {
+        return defaultDomain;
+    }
+
+    public void setDefaultUsername(String defaultUsername) {
+        this.defaultUsername = defaultUsername;
+        setTextWithDomain(defaultUsername);
+        isInitialized = true;
+    }
+
+    public String getDefaultUsername() {
+        return defaultUsername;
+    }
+
+    public String getUserName() {
+        int atIndex = getText().toString().indexOf('@');
```

Comment:
not sure how i feel about the '@' not being a variable. Also I can see its repeating at least two times. maybe it can be refactored but other than that good.